libbeat/outputs/redis/client.go

```diff
@@ -128,7 +128,10 @@ func (c *client) Publish(batch publisher.Batch) error {
 	if rest != nil {
```

Comment:
Question so I would undestand: what does publish only return event if an error occured?

Reply:
Now I see what is happening here.

Reply:
The publish functions do return the events not being published yet. Normally that's true if an error occured. We always return the error value, just in case.

Reply:
Thanks.